rtt_rosparam/README.md

```diff
@@ -22,53 +22,63 @@ policy (see below).
+* `get(name ,policy)` Attempt to get the property named `name` (or populates the properties of a named RTT sub-service)
+  from the ROS parameter namespace specified by `policy`.
+* `getRelative(name)`
+* `getAbsolute(name)`
+* `getPrivate(name)`
+* `getComponentPrivate(name)`
 
 ##### Operations for setting single properties
 
-* **setParam(ros_name, rtt_name)** Set the ROS param **ros_name** from the value in the RTT property **rtt_name**. Use leaders like `~` and `/` for private and absolute resolution.
-* **set(name, policy)** Attempt to set the property named **name** (or stores the properties of a named RTT sub-service)
-  in the ROS parameter namespace specified by **policy**.
-* **setRelative(name)**
-* **setAbsolute(name)**
-* **setPrivate(name)**
-* **setComponentPrivate(name)**
+* `setParam(ros_name, rtt_name)` Set the ROS param `ros_name` from the value in the RTT property `rtt_name`*. Use leaders like `~` and `/` for private and absolute resolution.
+* `set(name, policy)` Attempt to set the property named `name` (or stores the properties of a named RTT sub-service)
+  in the ROS parameter namespace specified by `policy`.
+* `setRelative(name)`
+* `setAbsolute(name)`
+* `setPrivate(name)`
+* `setComponentPrivate(name)`
+
+##### Operations to create properties linked to a ROS param
+
+* `addRosParamProperty_type_(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `Relative` namespace resolution context.
+* `addRosParamProperty_type_Relative(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `Relative` namespace resolution context.
+* `addRosParamProperty_type_Absolute(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `Absolute` namespace resolution context.
+* `addRosParamProperty_type_Private(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `Private` namespace resolution context.
+* `addRosParamProperty_type_ComponentPrivate(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `ComponentPrivate` namespace resolution context.
+* `addRosParamProperty_type_ComponentAbsolute(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `ComponentAbsolute` namespace resolution context.
+* `addRosParamProperty_type_ComponentRelative(name)` Adds a property of type `_type_` with name `name` to the owning task context which is linked to the ROS parameter with the same name in a `ComponentRelative` namespace resolution context.
```

Comment:
We agreed that the suggested behavior that replaces existing data sources is error-prone and should not be implemented, because the resulting behavior breaks method 1 above.